src/openrct2/ride/Vehicle.cpp

```diff
@@ -9858,3 +9855,41 @@ void Vehicle::Serialise(DataSerialiser& stream)
     stream << BoatLocation;
     stream << IsCrashedVehicle;
 }
+
+constexpr const LegacyAnimationParameters VehicleEntryDefaultAnimation[NumLegacyAnimationTypes] = {
+    { 0, 1, VehicleEntryAnimation::None },                  // None
+    { 1 << 12, 4, VehicleEntryAnimation::SteamLocomotive }, // Miniature Railway Locomotive
+    { 1 << 10, 2, VehicleEntryAnimation::SwanBoat },        // Swan Boat
+    { 1 << 11, 6, VehicleEntryAnimation::SimpleVehicle },   // Canoe
+    { 1 << 11, 7, VehicleEntryAnimation::SimpleVehicle },   // Rowboat
+    { 1 << 10, 2, VehicleEntryAnimation::SimpleVehicle },   // Water Tricycle
+    { 0x3333, 8, VehicleEntryAnimation::ObservationTower }, // Observation Tower
+    { 1 << 10, 4, VehicleEntryAnimation::SimpleVehicle },   // Mini Helicopter
+    { 1 << 11, 4, VehicleEntryAnimation::MonorailCycle },   // Monorail Cycle
+    { 0x3333, 8, VehicleEntryAnimation::MultiDimCoaster },  // Multi Dimension Coaster
+    { 24, 4, VehicleEntryAnimation::AnimalFlying },         // Animal Flying
+};
+
+const std::string VehicleEntryAnimationNames[VehicleEntryAnimation::Count] = {
+    "none",          "simpleVehicle",   "steamLocomotive",  "swanBoat",
+    "monorailCycle", "multiDimCoaster", "observationTower", "animalFlying",
+};
+
+uint8_t GetAnimationTypeFromString(std::string animationType)
```

Comment:
```suggestion
uint8_t GetAnimationTypeFromString(std::string_view animationType)
```

Reply:
Also the only place this is used is RideObject.cpp, better move it in there as a static function.

Reply:
Also this should result ```VehicleEntryAnimation``` and not some integer type, currently its unclear what this is even supposed to do, assuming thats what this function is doing currently?

Reply:
yes, it is intended to get `VehicleEntryAnimation` from string. I forgot that I could return `VehicleEntryAnimation` as a type